Создан класс квадрат

tfilex opened the pull request:
Предусмотрены поля: `координаты левого верхнего угла`, `длина стороны`;
а также следующие методы:
`- инициализация полей заданного класса;`

`- вывод на экран координат всех вершин квадрата;`

`- изменение длины стороны;`

`- изменение координат квадрата;`

`- вычисление периметра квадрата;`

`- вычисление площади квадрата;`

`- вычисление координат вершин квадрата, если он повернут на 90 градусов.`

tfilex committed:
Создан класс квадрат

MarinaPaley commented on square.h:
```diff
@@ -0,0 +1,187 @@
+
+	Square()
+	{
+		x1 = 0.0;
```
Пишите явно всегда this.
Лучше просто
![IMG_20210414_134032.jpg](https://user-images.githubusercontent.com/26978698/114697649-d99a0580-9d26-11eb-8d85-bec586ea5360.jpg)
Пример с конструктором

MarinaPaley reviewed (changes requested):
Сделайте деструктор по умолчанию
Конструктор копирования
Операторы присваивания, 
Добавьте explicit для конструктора
Уберите конструктор по умолчанию
Добавьте документацию к каждому методу

tfilex committed:
Сделано:

Деструктор по умолчанию,
Конструктор копирования,
Добавлен explicit для конструктора,
Убран конструктор по умолчанию,
Добавлена документация к каждому методу,
Реалезация перенесена в отдельный cpp,
Добавлена проверка длины через throw.

MarinaPaley commented on square.h:
```diff
@@ -0,0 +1,275 @@
+class Square
+{
+private:
+	double x1;
```
Документация по всем методам. полям  к классу

MarinaPaley commented on square.h:
```diff
@@ -0,0 +1,275 @@
+private:
+	double x1;
+	double y1;
+	double length;
```
сеттер стороны с проверкой

MarinaPaley commented on square.h:
```diff
@@ -0,0 +1,275 @@
+		* \param изменение длины
+		* \return изменённая длина
+	*/
+	double get_length_change(double Change_of_length);
```
почему переменная вдруг с большой буквы..
изменять длину нельзя. Когда задаете новую координу, смотрите, чтобы сторона не изменялась.
Сделайте структруру `Point` с логикой проверки на `==` двух точек. `abs(x1 - x2) <= numeric_limits<double>min() ...` обе координаты проверяйте

MarinaPaley commented on Square.cpp:
```diff
@@ -0,0 +1,65 @@
+
+Square::Square(double x1, double y1, double length)
+{
+	if (length > 0)
```
используйте сеттер

MarinaPaley commented on Square.cpp:
```diff
@@ -0,0 +1,65 @@
+double Square::get_vortex_y3() { return y1 - length; }
+double Square::get_vortex_y4() { return y1 - length; }
+
+double Square::get_length_change(double Change_of_length) { return length + Change_of_length; }
```
запретить изменение длины, но при этом разрешить изменение положения

tfilex committed:
Доделана документация по всем методам и полям;

удалены методы изменения длины;
структуру point делать не стал, по причине того, что это лишь излишнее усложнение программы.

MarinaPaley reviewed (commented):
Добавьте [оператор сдвига](https://en.cppreference.com/w/cpp/language/operators)

tfilex committed:
Добавлен оператор сдвига
